source/common/network/io_socket_handle_impl.cc

```diff
@@ -492,5 +492,18 @@ Event::FileEventPtr IoSocketHandleImpl::createFileEvent(Event::Dispatcher& dispa
 Api::SysCallIntResult IoSocketHandleImpl::shutdown(int how) {
   return Api::OsSysCallsSingleton::get().shutdown(fd_, how);
 }
+
+absl::optional<std::chrono::milliseconds> IoSocketHandleImpl::lastRoundTripTime() {
+  struct tcp_info ti;
+  socklen_t len = sizeof(ti);
+  if (!SOCKET_FAILURE(Api::OsSysCallsSingleton::get()
```

Comment:
It's up to you, but for efficiency sake you can likely determine ahead of time (even if constexpr function if TCP_INFO is defined) if this is supported, otherwise short circuit the call that will fail.

Reply:
Can you explain what you mean by this?

Reply:
See for example:

https://github.com/envoyproxy/envoy/blob/d0b8177f9682a44623dffff06f9d3f7c414627b0/source/common/network/io_socket_handle_impl.cc#L34-L40

And its use.

The idea is you can know ahead of time whether the platform supports the operation or not. So just switch on that (or use an ifdef here as is done elsewhere in this file) and either make the syscall on linux, or just directly return on the other platforms. You don't need to do the dance of making a syscall that you know will fail on other platforms.

Reply:
I think the suggestion is basically to `#ifdef TCP_INFO` this entire section, instead of redefining the `TCP_INFO` and making it always fail on platform without support.

Reply:
Right